rust_src/src/fns.rs

```diff
@@ -0,0 +1,43 @@
+extern crate libc;
+
+use std::ptr;
+use std::os::raw::c_char;
+use lisp::{LispSubr, PSEUDOVECTOR_AREA_BITS, PvecType, VectorLikeHeader, LispObject, SBYTES,
+           SSDATA, STRING_MULTIBYTE};
+use strings::STRINGP;
+use cons::NILP;
+
+static MIME_LINE_LENGTH: isize = 76;
+
+extern "C" {
+    fn make_unibyte_string(s: *const libc::c_char, length: libc::ptrdiff_t) -> LispObject;
+    fn base64_encode_1(from: *const libc::c_char, to: *mut libc::c_char, length: libc::ptrdiff_t,
+                       line_break: bool, multibyte: bool) -> libc::ptrdiff_t;
+}
+
+pub fn Base64EncodeString (string: LispObject, noLineBreak: LispObject) -> LispObject {
```

Comment:
This doesn't need to be `pub` as far as I can see.

Also, could you use the standard Emacs naming here? `Fbase64_encode_string`, `Sbase64_encode_string`.